src/core/evaluator.js

```diff
@@ -1551,8 +1550,14 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
       // 9.10.2
       var toUnicode = (dict.get('ToUnicode') || baseDict.get('ToUnicode'));
       if (toUnicode) {
-        properties.toUnicode = this.readToUnicode(toUnicode);
+        return this.readToUnicode(toUnicode)
+        .then(function(toUnicode) {
+          properties.toUnicode = toUnicode;
+          return properties;
+        });
       }
+
+      return new Promise(function (resolve, reject) {
```

Comment:
this does not look right, please create e.g. unicodeRead promise and set it to the result of readToUnicode or just Promise.resolve is !toUnicode case.